plugins/vlc-video/vlc-video-source.c

```diff
@@ -394,12 +394,115 @@ static int vlcs_audio_setup(void **p_data, char *format, unsigned *rate,
 	return 0;
 }
 
+static inline void vlc_split_media_path(const struct dstr *new_path,
+		struct dstr *path_only, struct dstr *options)
+{
+	int i;
+	const char *path = new_path->array;
+	bool quoting = false;
+	int options_at = 0;
+	int path_len;
+
+	for (i = 0; path[i]; ++i) {
+		if (path[i] == '"') {
+			quoting = !quoting;
+		}
+		if (!quoting && path[i+1] == ' ' && path[i+2] == ':') {
+			// ++i to be consistent with finished loop case
+			options_at = ++i;
+			break;
+		}
+	}
+	if (options_at) { // grab options
+		dstr_copy(options, new_path->array + options_at);
+	}
+	for(--i; path[i] == ' '; --i); // remove trailing spaces
+	path_len = i + 1;
+	dstr_ncopy_dstr(path_only, new_path, path_len);
+	dstr_replace(path_only, "\"", "");
+}
+
+static inline void vlc_apply_media_options(libvlc_media_t *media,
+		const char *options, const bool is_url)
+{
+	int i;
+	int j;
+	int k;
+	bool quoting = false;
+	int option_at = 0;
+	int option_len;
+	char *option_str = 0;
+	bool custom_network_caching = false;
+
+	if (!options)
+		return;
+	for (i = 0; options[i]; ++i) {
+		if (options[i] == '"') {
+			quoting = !quoting;
+		}
+		if (!option_at) {
+			if (options[i] == ':') {
+				// option begins
+				option_at = i;
+			}
+		} else if (options[i+1] == '\0' || (!quoting &&
+			options[i+1] == ' ' && options[i+2] == ':')) {
+			// option completed
+			option_len = i + 1 - option_at;
+			for (j = i; j != option_at; j--) {
+				// remove trailing spaces
+				if (options[j] == ' ') {
+					--option_len;
+				} else {
+					break;
+				}
+			}
+			option_str = malloc((option_len + 1) *
```

Comment:
Style:  Lines 460-461 can actually be one line since they do not exceed 80 characters.

Instead of:
```
			option_str = malloc((option_len + 1) *
				sizeof(char));
```

use:
```
			option_str = malloc((option_len + 1) * sizeof(char));
```